sympy/solvers/bivariate.py

```diff
@@ -302,6 +305,9 @@ def _solve_lambert(f, symbol, gens):
         raise NotImplementedError('%s does not appear to have a solution in '
             'terms of LambertW' % f)
 
+    soln = [s for s in soln if
+              checksol(f, {symbol: s}) is not False]
```

Comment:
I know, I added because of this test case
```
p = symbols('p', positive=True)
eq = 4*2**(2*p + 3) - 2*p - 3
assert _solve_lambert(eq, p, _filtered_gens(Poly(eq), p)) == [
    -S(3)/2 - LambertW(-4*log(2))/(2*log(2))]
```